File: examples/cluster/example_mixed_clustering.R
zfaithful <- scale(faithful)
# hclust
d    <- dist(zfaithful)
cl1  <- hclust(d, method="ward.D2")
memb <- cutree(cl1, 3)
# kmeans
groupm  <- aggregate(zfaithful, list(memb), mean)
centers <- cbind(groupm$eruptions, groupm$waiting)
cl2 <- kmeans(zfaithful, centers = centers)
# compare results (confusion matrix)
tab <- table(memb, cl2$cluster)
tab
# match measures
psych::cohen.kappa(tab)
DescTools::ContCoef(tab)
DescTools::Phi(tab)
DescTools::CramerV(tab)

File: examples/cluster/example_mixed_clustering.py
import numpy as np, pandas as pd
from sklearn.preprocessing import StandardScaler
from sklearn.cluster import KMeans
from scipy.cluster.hierarchy import linkage, fcluster
from scipy.spatial.distance import pdist
from sklearn.metrics import confusion_matrix
zfaithful = StandardScaler().fit_transform(r["faithful"])
# Hierarchical clustering
d = pdist(zfaithful)
linkage = linkage(d, method='ward')
memb    = fcluster(linkage, t=3, criterion='maxclust')
# K-means, initial cluster centers from hierarchical clustering
groupm = pd.DataFrame(zfaithful).groupby(memb).mean().values
kmeans = KMeans(n_clusters=3, init=groupm, n_init=1, random_state=42)
kmeans.fit(zfaithful)
tab = confusion_matrix(memb, kmeans.labels_)
print(tab)
